docs/design/principles.md

```diff
@@ -1,54 +1,47 @@
 # ICU4X Design Principles & Decisions
 
-## Guiding principles
-
 These principles are not cast in stone, but are strong guidelines for developers.
 
-### Immutable objects
+## i18n Best Practices
+
+Above all, ICU4X must provide modern, standards-compliant APIs that encourage best i18n practices and produce correct results for all languages and locales. No language or locale should be at a structural disadvantage.
+
+## Code Style
 
-No object is to be mutated after creation.
+All ICU4X code must conform to the [style guide](../process/style_guide.md), including the following:
 
-### No I/O in the core library
+### Safety
 
-Leave I/O to the wrapper or clients. Pass data as reference to the methods.
+All standard Rust practices regarding mutability, lifetimes and safety must be followed.
```

Comment:
Do we want to list some of them (or is there a good doc to point to)? I ask because Immutability was removed from the doc and I am not sure if it's covered by this sentence.

Reply:
The way Rust deals with immutability is different than how Java deals with it.  For example, in Java, there is mutable `StringBuilder` and immutable `String`; in Rust, there is just `String`, and to mutate it, you obtain a mutable reference, which the borrow checker verifies to be safe (only one thread can mutate at a time, and only while no one else is borrowing).  I removed "Immutability" and replaced it with "standard Rust practices regarding mutability" to try to express that better.